janitor/functions/collapse_levels.py

```diff
@@ -72,6 +72,8 @@ class  max_speed    type
     if not isinstance(df.columns, pd.MultiIndex):
         return df
 
+    df = df[:]
```

Comment:
@samukweku what does this syntax do for us here? Should it be `df = df.copy()` instead?

Reply:
We are mutating the columns in this case, so a slice suffices. Pandas creates a new index altogether, which does not affect the original dataframe.